### ISSUE TITLE
Cannot compile when Typescript namespaces contain keywords

### ISSUE THREAD

**Issue by haynesgt** [vojtechhabarta/typescript-generator#135]
Typescript keywords in Java packages create errors (error TS1003: Identifier expected.)

The following are invalid in Typescript namespaces but can appear in java package names:

[ 'debugger',
  'delete',
  'export',
  'function',
  'in',
  'typeof',
  'var',
  'with' ]

E.g. here is a valid java package.

package debugger.delete.export.function.in.typeof.var.with;

Here are some invalid namespaces:

namespace a.debugger.c { }
namespace a.delete.c { }
namespace a.export.c { }
namespace a.function.c { }
namespace a.import.c { }
namespace a.in.c { }
namespace a.instanceof.c { }
namespace a.typeof.c { }
namespace a.var.c { }
namespace a.with.c { }

**Comment by vojtechhabarta** [comment]
Thanks for reporting this.

I think invalid namespace identifiers are first two groups from [Reserved Words](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#221-reserved-words) (`break` ..., `implements` ...).

What do you think we should do about this? Prefix these keywords with underscore (`_break`) or make first letter uppercase (`Break`)?

**Comment by vojtechhabarta** [comment]
Aha, your list of invalid namespace identifiers is not just example but it is the first group of reserved words minus Java reserved words.

**Comment by haynesgt** [comment]
Regarding solutions, both prefixes and case changes could conflict with other packages. I feel like the best option would be to add a prefix that is underscore by default (`_export`), but with an option (`<reservedWordPrefix>`) to let a user change it. I don't like uppercase namespaces because they look like classes.

**Comment by vojtechhabarta** [comment]
Yes they can conflict but it is not much probable that single project contains two packages that differ only in one `_` character and both having class with the same simple name. So I don't want to add new configuration setting for this prefix (at least for now). There are possibilities how to resolve these conflicts (for example using `customTypeNaming` parameter).

I preferred uppercase namespace (it is common in TypeScript and C#) but since you prefer underscore prefix I used this prefix 😄

**Comment by vojtechhabarta** [comment]
Fix released: [v1.24.317](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v1.24.317).
Could you please try it?

**Comment by haynesgt** [comment]
It works. I can compile without errors.

Thanks!
